src/util/simplify_expr.cpp

```diff
@@ -2270,9 +2271,27 @@ bool simplify_exprt::simplify_node(exprt &expr)
   else if(expr.id()==ID_good_pointer)
     result=simplify_good_pointer(expr) && result;
   else if(expr.id()==ID_div)
-    result=simplify_div(expr) && result;
+  {
+    if(expr.type().id() == ID_floatbv)
+    {
+      adjust_float_expressions(expr, ns);
+      result = simplify_rec(expr);
+    }
+    else
+    {
+      result = simplify_div(expr) && result;
```

Comment:
Why not move the `if(expr.type().id() == ID_floatbv)` inside `simplify_div` etc?